Vagrantfile

```diff
@@ -0,0 +1,250 @@
+# Vagrantfile to spin up a development VM using UTM
```

Comment:
Is this needed when we have nix shell and nixos shell?

I mean we have to maintain multiple environments, and the chance of breaking one of them later on is high

Reply:
Using Vagrant allows developers working on non-Linux machines (e.g. me ;-) to work on the project. nix-shell does not work on my machine, and I also was unable to spin up a nixos VM.

Reply:
CC: @tiagolobocastro

Reply:
Yeah not sure, we'd need someone who knows vagrant to maintain it working
CC @abhilashshetty04 @dsharma-dc